src/main/java/net/minecraftforge/fml/common/AutomaticEventSubscriber.java

```diff
@@ -79,7 +79,9 @@ public static void inject(ModContainer mod, ASMDataTable data, Side side)
                     }
                     FMLLog.log.debug("Registering @EventBusSubscriber for {} for mod {}", targ.getClassName(), mod.getModId());
                     Class<?> subscriptionTarget = Class.forName(targ.getClassName(), false, mcl);
-                    MinecraftForge.EVENT_BUS.register(subscriptionTarget);
+                    ModAnnotation.EnumHolder bus = (ModAnnotation.EnumHolder)targ.getAnnotationInfo().get("bus");
+                    if (bus != null) ForgeBusType.valueOf(bus.getValue()).getBus().register(subscriptionTarget);
```

Comment:
If the bus is null, an exception should be thrown. If a mod is trying to register to a bus that they do not specify, it is a problem in their code. Silently falling back to the main event bus is not the way this should be handled.